components/ProductCard.js

```diff
@@ -0,0 +1,117 @@
+import styled from 'styled-components';
+import QuantitySelector from './QuantitySelector';
+import { useState } from 'react';
+
+export default function ProductCard({ product }) {
+  const [showDetails, setShowDetails] = useState();
+
```

Comment:
```suggestion
const { id, image, name, ... } = product
```
consider to destructure